src/prx/main.py

```diff
@@ -387,10 +387,32 @@ def format_flat_rows(row):
     sat_states["tropo_delay_m"] = tropo_delay_m
     # Compute time_system_corr_dict somewhere in your code before using it
     sat_states["constellation"] = sat_states["satellite"].str[0]
-    time_system_corr_dict = helpers.parse_rinex_nav_file(rinex_3_ephemerides_file)  # Define your dictionary here
-    icb_all_constellations = helpers.compute_icb_all_constellations(time_system_corr_dict)
+    sat_states["time_of_reception_in_receiver_time_weeks_seconds"] = sat_states.apply(
+        lambda row: helpers.timedelta_2_weeks_and_seconds(
+            row.time_of_reception_in_receiver_time
+            - constants.system_time_scale_rinex_utc_epoch["GPST"]
+        )[1],
+        axis=1,
+    )
+    sat_states["time_of_reception_in_receiver_time_weeks"] = sat_states.apply(
+        lambda row: helpers.timedelta_2_weeks_and_seconds(
+            row.time_of_reception_in_receiver_time
+            - constants.system_time_scale_rinex_utc_epoch["GPST"]
+        )[0],
+        axis=1,
+    )
+    time_system_corr_dict = helpers.parse_rinex_nav_file(rinex_3_ephemerides_files)  # Define your dictionary here
```

Comment:
This does not create one dictionary per file, when there are multiple NAV files. You can look at how it is done for the IONO CORR downstream in this same function.

A dictionary of dictionaries is created, with the string 'yyyyddd' as a key. Then, the correction is computed on the right dictionary, based on the day of the time of reception.

You can try to adapt the same strategy for ICB.